python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/utils/asset_scenario_state.py

```diff
@@ -155,8 +155,9 @@ def with_all_eager(self, max_materializations_per_minute: int = 1) -> Self:
             )
         )
 
-    def with_current_time(self, time: str) -> Self:
-        return dataclasses.replace(self, current_time=pendulum.parse(time))
+    def with_current_time(self, time: Union[str, datetime.datetime]) -> Self:
```

Comment:
why is this change necessary?

Reply:
just QoL -- nice to be able to pass in either a string or a datetime to this sort of method

Reply:
(specifically, I do that on line 36 of missing_tests.py)